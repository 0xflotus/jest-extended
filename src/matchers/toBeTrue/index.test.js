import each from 'jest-each';

import matcher from './';

expect.extend(matcher);

describe('toBeTrue', () => {
  it('passes when given true', () => {
    expect(true).toBeTrue();
  });

  it('fails when not given true', () => {
    expect(() => expect(false).toBeTrue()).toThrowErrorMatchingSnapshot();
  });
});

describe('.not.toBeTrue', () => {
  each([[false], [''], [0], [{}], [[]], [() => {}], [undefined], [null], [NaN]]).it(
    'passes when not given true: %s',
    given => {
      expect(given).not.toBeTrue();
    }
  );

  it('fails when given true', () => {
    expect(() => expect(true).not.toBeTrue()).toThrowErrorMatchingSnapshot();
  });
});
