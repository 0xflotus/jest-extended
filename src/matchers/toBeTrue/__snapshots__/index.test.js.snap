// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`.not.toBeTrue fails when given true 1`] = `
"<dim>expect(</><red>received</><dim>).not.toBeTrue(</><dim>)</>

Expected value to not be true received:
  <red>true</>"
`;

exports[`.toBeTrue fails when not given true 1`] = `
"<dim>expect(</><red>received</><dim>).toBeTrue(</><dim>)</>

Expected value to be true:
  <green>true</>
Received:
  <red>false</>"
`;
